src/nvim/vim.h

```diff
@@ -316,8 +316,60 @@ enum { FOLD_TEXT_LEN = 51 };  //!< buffer size for get_foldtext()
 #define LOWEST_WIN_ID 1000
 
 // BSD is supposed to cover FreeBSD and similar systems.
-#if (defined(BSD) || defined(__FreeBSD_kernel__)) && defined(S_ISCHR)
-# define OPEN_CHR_FILES
+#if (defined(BSD) || defined(__FreeBSD_kernel__)) \
+    && (defined(S_ISCHR) || defined(S_IFCHR))
+#define OPEN_CHR_FILES
+#endif
+
+// stat macros
+#ifndef S_ISDIR
```

Comment:
These were previously defined in `src/nvim/os/os_defs.h` but moved to `src/nvim/os/win_defs.h` in  07265d22. That was a mistake. Move these ifndefs to `os_defs.h` and remove the duplicates in `win_defs.h`.